packages/react-components/src/components/chart/hooks/useChartSetOptionSettings.ts

```diff
@@ -0,0 +1,25 @@
+import { useMemo, useRef } from 'react';
+import { DataStream } from '@iot-app-kit/core';
+
+const mapDatastreamsDeps = (datastreams: DataStream[]) =>
+  JSON.stringify(datastreams.map((datastream) => datastream.id));
+
+export const useChartSetOptionSettings = (datastreams: DataStream[]) => {
+  const datastreamsDeps = mapDatastreamsDeps(datastreams);
+  const datastreamsDepsRef = useRef(datastreamsDeps);
+
+  return useMemo(() => {
+    /**
+     * if the datastreams change update echarts using the replaceMerge stratgey
+     * so that it is ensured that orphaned data points are removed.
+     *
+     * see setOption api for more information on settings
+     * https://echarts.apache.org/en/api.html#echartsInstance.setOption
+     *
+     * NOTE: we can use this hook to refactor trend cursor set option also
```

Comment:
Maybe settings should not be in 1 place, I think even this hook might not be exactly right. but I think we can use a similar strategy in TCs so that all of our hooks only return state instead of calling setOptions internally. I think that would be a cleaner pattern.

Reply:
only return state instead of calling setOptions internally --> thats how TC pattern is , only exception is the delete TC. Again, features should be calling setOptions as an exception, not everytime, which is also the pattern in TC